torch/csrc/jit/tensorexpr/eval.h

```diff
@@ -318,6 +318,10 @@ AT_FORALL_SCALAR_TYPES_AND(Half, TYPE_CASE);
         break;
 AT_FORALL_SCALAR_TYPES_AND(Half, TYPE_CASE);
 #undef TYPE_CASE
+      case ScalarType::Bool:
+        LOG(WARNING) << "Warning doing binary operation on Bool";
```

Comment:
Was this warning left over from debugging?  I assume we don't actually want to log from here.

Reply:
No I just figured actually trying to multiply two bools was unlikely enough people would want to know about it. I'll remove it but now we'll silently do arithmetic on chars and then treat them as bools.

Reply:
Hm, yeah bool is weird, I just tried out a bunch of binary ops in the torch interp:
```
] python
Python 3.8.1 (default, Jan  8 2020, 22:29:32) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> x=torch.tensor([True, False])
>>> x
tensor([ True, False])
>>> x=torch.tensor([True, False], dtype=torch.bool)
>>> x
tensor([ True, False])
>>> x=torch.tensor([0,1], dtype=torch.bool)
>>> x
tensor([False,  True])
>>> y=torch.tensor([1,0], dtype=torch.bool)
>>> y
tensor([ True, False])
>>> x*y
tensor([False, False])
>>> x^y
tensor([True, True])
>>> x&y
tensor([False, False])
>>> x|y
tensor([True, True])
>>> x<<y
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: "lshift_cpu" not implemented for 'Bool'
>>> x+y
tensor([True, True])
>>> x/y
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: "div_cpu" not implemented for 'Bool'
>>> x-y
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: Subtraction, the `-` operator, with two bool tensors is not supported. Use the `^` or `logical_xor()` operator instead.
>>> 
```

idk how far down the bool rabbit hole we should go... maybe worth seeing what the current fuser does with them.

Reply:
If I'm reading this right that means, mul -> AND, add -> OR, sub, div and shifts -> Error and bitwise works fine.

That could just be a graph transformation?